### ISSUE TITLE
Feature Request: Automatic XML Import from URI

### ISSUE THREAD

**Issue by darkpixel** [mRemoteNG/mRemoteNG#2333]
I would love to see a feature in mRemote where it would automatically load/import an XML config file on startup.

This would be helpful for provisioning mRemote for lots of users.

An example environment would be a call center where users needs to connect out to various offices to pull up records.

Rather than reminding everyone to "delete their mRemote config and then re-import it from the usual location", I would love to tell mRemote to always automatically load connections from ```https://example.tld/mremote.xml```

Even if it were a "just a command line arg", we could deploy our own shortcut to start mremote.  i.e. ```c:\progra....\mremote.exe -loadconfig https://example.tld/mremote.xml```

I suppose you could get fancier by creating a file handler for ```.mrng``` files and you could just distribute these XML snippets to users and they could double-click to import--similar to a .reg file.

**Comment by Kvarkas** [comment]
what will be posible with profiles implementation, but also its a lot of questions of security for such way - file should be encrypted, and user should have access to it, in my world (where i am working at the moment) we control credentials with cyberark, so user never know password. mremote have only connection details - then call cyberArk for credentials and login into system. On next login same credentials will not work if password or user name (or both) has been changed :)

**Comment by simonai1254** [comment]
I don't see the direct use case when one could use a database for the connection profiles (current implementation works but requires improvement). The database would dynamically update the settings everywhere without the need of downloading config files on startup. I see this kind of distribution with proxy configurations and IMHO it's quite a pain to maintain...

On the other hand, the ability to import connections with a double click would be great, maybe also allowing for one-time use (like .rdp files created by tools like CyberArk or downloaded from RDWeb Servers).

**Comment by darkpixel** [comment]
I don't want to:
1. Provision a database in the cloud (why spend $15/mo?) and make it accessible from everywhere on the internet as users roam all over the place and don't have static IPs.
2. Walk users through picking "mysql" from the dropdown, entering "mremote-connections-do-user-134425-0.b.db.ondigitalocean.com" in the hostname, entering a database name, username, password, and checking the 'read only' button.
3. Spend 15 minutes telling each of 300 users the database password is 'tathie6liNg9Ieyaib1ongeis3dohd6u' (no, that's 't' as in 'tango', not 'd' as in 'delta') because the password shouldn't be easily guessed by random people who might want to gather info on our RDP connections. (or connecting to 300 machines and provisioning it manually myself)
4. Be restricted from changing a password due to it requiring a significant time investment to update on 300 machines

I haven't touched anything in the Microsoft dev world in a decade now, but I would think adding command line arg or a setting would be fairly quick.

**Comment by Kvarkas** [comment]
no, you should not, but it should be yours design not our how save you want to be :) we only give you a right way to do it, but we cant restrict you from using credentials in open form in flat file :) but as i mentioned idea is that once profiles will be implemented you my set from where to load it and how (it could be encrypted or not - up to you) and idea is that you as admin configure the settings - users dont need to know even from where credentilas are comming - you just share them connection profile that franckly speaking will be a collection of other links and settings and mremoteng will know how to handle it, so yes if you want to send it as line arg

**Comment by robertpopa22** [comment]
**Fix available for testing**

**Commit:** `0b73cea5` (fork `robertpopa22/mRemoteNG`, branch `main`)
**What changed:** Feature Request: Automatic XML Import from URI

A beta build (v1.81.0-beta.4) including this fix is available from the fork's Releases page.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*
